src/Game.py

```diff
@@ -32,4 +35,82 @@ def initGame(config1, config2, size):
     else:
         currentPlayer = PLAYER_2
 
-    return boards, currentPlayer
+    return gameState, currentPlayer
+
+def decrementTargetFleetLife(boards, attacker):
+    boardsCopy = deepcopy(boards)
+
+    if attacker == PLAYER_1:
+        boardsCopy[PLAYER_2]["life"] -= 1
+    else:
+        boardsCopy[PLAYER_1]["life"] -= 1
+
+    return boardsCopy
+
+def gameIsWon(boards, player):
+    gameIsWon = True
+
+    # check all of the ships coords on the board and look if it contains SUCCESS
+    for ship in boards[playerOpponent(player)]["ships"]:
+        for coord in ship:
+            if not isShipPartHit(coord, boards, playerOpponent(player)):
+                gameIsWon = False
+                break
+
+    return gameIsWon
+
+def playerOpponent(player):
+    if player == PLAYER_1:
+        return PLAYER_2
+    else:
+        return PLAYER_1
+
+def shoot(coord, gameState, currentPlayer):
+
+    if (currentPlayer == PLAYER_1):
+        updatedGameState, hit = updateBoardsAndHit(coord, gameState, PLAYER_1, PLAYER_2)
+    else:
+        updatedGameState, hit = updateBoardsAndHit(coord, gameState, PLAYER_2, PLAYER_1)
+
+    shipSunk = False
+
+    if hit:
+        # check entire ship:
+        # 1 find ship line in opponent ships
+        if currentPlayer == PLAYER_1:
+            ships = gameState[PLAYER_2]["ships"]
+        else:
+            ships = gameState[PLAYER_1]["ships"]
+
+        shipLine = []
+
+        for ship in ships:
+            for partCoord in ship:
+                if partCoord == coord:
+                    shipLine = ship
+                    break
+
+        # 2 check each coord of the ship with isShipPartHit
```

Comment:
I don't understand this part

Reply:
After having found the ship to verify, I need to look into the opponent board and test if all the coordinates of the ship are hit. This tells me if the ship has sunk.